lib/Runtime/Language/InterpreterLoop.inl

```diff
@@ -166,7 +166,7 @@ Var Js::InterpreterStackFrame::INTERPRETERLOOPNAME()
     }
 
     Assert(this->returnAddress != nullptr);
-    AssertMsg(m_arguments == NULL || Js::ArgumentsObject::Is(m_arguments), "Bad arguments!");
```

Comment:
Instead of removing this assert, you should be checking if there is arguments usage in the function body, if so then check the validity of it. 
You can make use of m_functionBody to see if the function body has arguments object or not.